torch/distributed/_tensor/op_schema.py

```diff
@@ -121,26 +126,24 @@ def output_ndim(self):
 class TupleStrategy(StrategyType):
     """
     TupleStrategy represents the output strategy of this op is a tuple
-    of strategy, i.e. If the output of this op is a tuple of tensors, we should
-    return a TupleStrategy that contains a tuple of OpStrategy.
+    of strategy, i.e. If the output of this op is a tuple of tensors or list of tensors
+    with possibly different placement strategies, we should return a TupleStrategy that
+    contains a tuple of OpStrategy.
 
-    NOTE: if the output of the op is a List[Tensor], it's likely we should return
-    OpStrategy directly in all cases.
+    NOTE: if the output of the op is a List[Tensor] and they share the same placement
+    strategy, then we should return a single OpStrategy instead of a TupleStrategy
```

Comment:
Why do we want to minimize the size of output strategy? Any performance regression if we returned a list of identical strategies instead of one OpStrategy for a list of tensors??

Reply:
I documented this so that we won't confuse with whether to use OpStrategy or TupleStrategy. If it's a placement that every output of list have the same strategy, we should use OpStrategy for simplicity reason and avoid any perf concerns, I don't think it would regress the perf but it would be good to keep TupleStrategy to only be used in certain cases.